dspace-api/src/main/java/org/dspace/authenticate/ShibAuthentication.java

```diff
@@ -509,19 +510,17 @@ public String loginPageURL(Context context, HttpServletRequest request, HttpServ
             int port = request.getServerPort();
             String contextPath = request.getContextPath();
 
-            String returnURL = request.getHeader("Referer");
-            if (returnURL == null) {
-                if (request.isSecure() || forceHTTPS) {
-                    returnURL = "https://";
-                } else {
-                    returnURL = "http://";
-                }
-                returnURL += host;
-                if (!(port == 443 || port == 80)) {
-                    returnURL += ":" + port;
-                }
+            String redirectUrl = null;
+            if (request.getHeader("Referer") != null && StringUtils.isNotBlank(request.getHeader("Referer"))) {
+                redirectUrl = request.getHeader("Referer");
+            } else if (request.getHeader("X-Requested-With") != null
+                    && StringUtils.isNotBlank(request.getHeader("X-Requested-With"))) {
+                redirectUrl = request.getHeader("X-Requested-With");
             }
 
+            String returnURL = ConfigurationManager.getProperty("dspace.baseUrl") + "/api/authn/shibboleth"
```

Comment:
Please note that this property it's about to be changed to: `dspace.server.url`

Reply:
Yes, this config now needs to be updated to `dspace.server.url`, as #2657 was merged